lib/internal/streams/whatwg.js

```diff
@@ -0,0 +1,4233 @@
+      writable: this.writable,
+      backpressure: this[kState].backpressure,
+    });
+  }
+
+  get [SymbolToStringTag]() { return 'TransformStream'; }
+
+  [kTransfer]() {
+    const {
+      readable,
+      writable,
+    } = this[kState];
+    if (readable.locked) {
+      throw new DOMException(
+        'Cannot transfer a locked ReadableStream',
+        'DataCloneError');
+    }
+    if (writable.locked) {
+      throw new DOMException(
+        'Cannot transfer a locked WritableStream',
+        'DataCloneError');
+    }
+    return {
+      data: {
+        readable,
+        writable,
+      },
+      deserializeInfo: 'internal/streams/whatwg:TransferedTransformStream'
+    };
+  }
+
+  [kTransferList]() {
+    return [ this[kState].readable, this[kState].writable ];
+  }
+
+  [kDeserialize]({ readable, writable }) {
+    this[kState].readable = readable;
+    this[kState].writable = writable;
+  }
+}
+
+class TransferedTransformStream extends JSTransferable {
+  constructor() {
+    super();
+    this[kState] = {
+      readable: undefined,
+      writable: undefined,
+      backpressure: undefined,
+      backpressureChange: {
+        promise: undefined,
+        resolve: undefined,
+        reject: undefined,
+      },
+      controller: undefined,
+    };
+  }
+}
+
+internalExtend(TransferedTransformStream, TransformStream);
+
+class TransformStreamDefaultController {
+  constructor() {
+    throw new ERR_ILLEGAL_CONSTRUCTOR();
+  }
+
+  /**
+   * @readonly
+   * @type {number}
+   */
+  get desiredSize() {
+    const {
+      stream,
+    } = this[kState];
+    const {
+      readable,
+    } = stream[kState];
+    const {
+      controller: readableController,
+    } = readable[kState];
+    return readableStreamDefaultControllerGetDesiredSize(readableController);
+  }
+
+  /**
+   * @param {any} chunk
+   */
+  enqueue(chunk) {
+    transformStreamDefaultControllerEnqueue(this, chunk);
+  }
+
+  /**
+   * @param {any} reason
+   */
+  error(reason) {
+    transformStreamDefaultControllerError(this, reason);
+  }
+
+  terminate() {
+    transformStreamDefaultControllerTerminate(this);
+  }
+
+  [kInspect](depth, options) {
+    return customInspect(depth, options, 'TransformStreamDefaultController', {
+      stream: this[kState].stream,
+    });
+  }
+
+  get [SymbolToStringTag]() { return 'TransformStreamDefaultController'; }
+}
+
+class InternalTransformStreamDefaultController {}
+
+class ByteLengthQueuingStrategy {
+  /**
+   * @param {{
+   *   highWaterMark : number
+   * }} init
+   */
+  constructor(init = {}) {
+    validateObject(init, 'init');
+    const {
+      highWaterMark
+    } = init;
+    if (typeof highWaterMark !== 'number') {
+      throw new ERR_INVALID_ARG_TYPE(
+        'init.highWaterMark',
+        'number',
+        highWaterMark);
+    }
+    this[kState] = {
+      highWaterMark,
+    };
+    this.size =
+      FunctionPrototypeBind(
+        ByteLengthQueuingStrategy.prototype.size,
```

Comment:
should this be cached, in case someone's mutated the prototype function prior to construction?

Reply:
This makes it easier/possible to use destructuring while retaining the `this`. It's not perfect but it works.

Reply:
that's not what i mean, the bind is fine - it's that you're looking up `.size` on the mutable `ByteLengthQueuingStrategy.prototype` at runtime, which means that it could have been deleted or replaced.

Reply:
I've changed it now to refer to this.size. Binding to a mutable version of the function is fine given the purpose here